Profiler/utils/functionID_set/functionId_set.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+#include <corprof.h>
+#include <limits.h>
+
+/// <summary>
+/// Set that can only contain functionIDs, which are just unsigned ints.
+/// This is based on an array and is a lot faster than the default set implementation of the standard library for this use case.
+/// </summary>
+class functionID_set
+{
+private:
+	const unsigned int default_size = 2'097'152;
+	static const unsigned int rotation_mask = (CHAR_BIT * sizeof(default_size) - 1);
+
+	unsigned int current_size = default_size;
+	unsigned int num_elements = 0;
+	unsigned int max_elements = default_size / 2;
+	FunctionID* set = new FunctionID[default_size]{ 0 };
+
+
+	// Predetermined values for xor operation to place in different spots
+	unsigned int xor_values[10] = { 2134038170, 2362107340, 3229546752, 1302939050, 200405764, 79516981, 2052331209, 3415124361, 940592490, 430981309 };
+
+	/// <summary>
+	/// Updates the size of the array and reinserts the element into the new array.
+	/// </summary>
+	void adjust_size() {
+		current_size *= 2;
```

Comment:
can this yield to overflows if done unchecked?

Reply:
then maybe just throw an exception as well? I just fear that this way we end up with corrupted memory.

Reply:
I did a bit of testing and when it cannot allocate the memory (at least when `new` is used), it throws std::bad_alloc and stops the application. That should be enough, I guess.